### PR TITLE
Fix API page response format for url and path

### PR DESCRIPTION
**Pull request by melegiul**
### Short description

Fix API page response format for url and path

### Proposed changes

- Prepends domain of backend to url translation path, reads domain from settings value `BASE_URL` (or is `WEBAPP_URL` expected here?)
- Sets path to permalink surrounded by slashes

Also updates url and path for these objects (included in page/children response):
1. parent
2. available_languages (path was not present before)

### Resolved issues

Fixes: #928

### CODE REVIEW

**Comment by svenseeberg** [comment]
The url and path variables look good now. I can see that there are more pages returned in https://cms-django-dev.integreat-app.de/albdonaukreis/en/wp-json/extensions/v3/children?depth=1 than in https://cms-dev.integreat-app.de/albdonaukreis/en/wp-json/extensions/v3/children?depth=1. I guess that the depth parameter is used differently. Also, the app currently breaks with the response. I asked the app team for help

File: src/api/v3/pages.py [hunk 1]
```diff
@@ -29,17 +29,19 @@ def transform_page(page_translation):
             "id": page_translation.page.parent.id,
```
Comment (svenseeberg) [inline src/api/v3/pages.py:29]: Until now the API returned a `0` if there is no page parent. With Django, the response currently contains `null`. We need to change this as well.

**Review by svenseeberg — CHANGES_REQUESTED** [review]

**Comment by melegiul** [comment]
> The `url` and `path` variables look good now. I can see that there are more pages returned in https://cms-django-dev.integreat-app.de/albdonaukreis/en/wp-json/extensions/v3/children?depth=1 than in https://cms-dev.integreat-app.de/albdonaukreis/en/wp-json/extensions/v3/children?depth=1. I guess that the depth parameter is used differently. Also, the app currently breaks with the response. I asked the app team for help.

I was wondering about that behavior, too. cms-django-dev with depth=0 (no children) seems to return same response (9 elements) like cms-dev with depth=1 (with immediate childern), same seems to continue for other depth values:
- cms-django-dev with depth=1: 49 elements
- cms-dev with depth=2: 50 elements

Maybe related to WP-45 issue?

**Comment by svenseeberg** [comment]
I think the difference is the following: our Django CMS assumes that the default depth is 0 while WordPress assumes the default depth is 1. Also the interpretation of the values may differ. For WP a depth of 1 means the page itself plus its direct children. In case of a missing `url` parameter that means some sort of virtual root element plus its direct children, which corresponds to the first level of pages. If a page is provided, the page itself plus its children are returned. That may explain the "of by one" problem.

**Review by svenseeberg — APPROVED** [review]

**Pull request merged**